client/my-sites/stats/utils.js

```diff
@@ -129,3 +131,38 @@ export const getChartRangeParams = ( startDate, fromPeriod ) => {
 		chartPeriod,
 	};
 };
+
+/**
+ * Get the preview URL for a post.
+ * When a site has a mapped domain, ensures the preview URL uses the unmapped domain
+ * to avoid potential SSL/proxy issues during preview.
+ * @param {Object} state - The Redux state object
+ * @param {number} siteId - The site ID
+ * @param {number} postId - The post ID
+ * @returns {string|null} The preview URL with appropriate domain, or null if unavailable
+ */
+export const getMappedPreviewUrl = ( state, siteId, postId ) => {
+	const basePreviewUrl = getPostPreviewUrl( state, siteId, postId );
+	if ( ! basePreviewUrl ) {
+		return basePreviewUrl;
+	}
+
+	const site = getSite( state, siteId );
+	const shouldUseUnmappedDomain = site?.options?.is_mapped_domain && site?.options?.unmapped_url;
+
+	if ( ! shouldUseUnmappedDomain ) {
+		return basePreviewUrl;
+	}
+
+	try {
+		const previewUrl = new URL( basePreviewUrl );
+		const unmappedUrl = new URL( site.options.unmapped_url );
+
+		previewUrl.protocol = unmappedUrl.protocol;
+		previewUrl.host = unmappedUrl.host;
```

Comment:
The existing [getPostPreviewUrl](https://github.com/Automattic/wp-calypso/blob/c9304b1226c442f1aecfb9d22542609546cdbac0/client/state/posts/selectors/get-post-preview-url.js#L46-L50) selector also attempts to use the `unmapped_url`.

This utility is added because I have noticed a domain on WordPress.com could be example.com but the WordPress admin URL can be example.wordpress.com, for example this happens on jetpack.com

This utility forces the use the `example.wordpress.com` domain which is avoids the preview error due to `X-Frame-Options` error